cumulusci/oauth/tests/test_salesforce.py

```diff
@@ -163,6 +165,7 @@ def test_oauth_flow_error_from_auth(self):
         # wait for thread to complete
         t.join()
 
+    @mock.patch("time.sleep", time.sleep)
```

Comment:
These patches shouldn't be needed. There is a fixture which automatically patches time.sleep for all tests: https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/conftest.py#L16

Reply:
This is an UN-patch.

Your fixture confused the heck out of me and I don't think it makes sense in these cases. There is an explicit time.sleep() in the test cases in this file and turning time.sleep into a no-op doesn't make sense if the tests themselves are relying on the real time.sleep. I think my algorithm is probably resistant to the patching now (it wasn't before) but still, the patching causes more problems than it solves in this case, IMO.

Reply:
Oh I see, hmm. The point of my fixture was to avoid slowing down the tests when they test code that includes a sleep just to avoid polling too frequently, and it did have a real impact on test performance. But I hadn't noticed that there are also sleeps within test code. Would it help at all if the fixture did a real sleep with a shortened interval instead of a no-op?

Reply:
I think it would be worth experimenting with a time.sleep that sleeps for 1/10 or 1/20 of the time but it should be a separate PR. I tweaked my algorithm so it doesn't presume that time.sleep(1) really sleeps for 1 second so it doesn't matter so much to me anymore, but the (pre-existing) tests themselves do try to use sleep to reduce their busy-waitiing.